hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2137,6 +2143,36 @@ func (r *HostedClusterReconciler) reconcileControlPlaneOperator(ctx context.Cont
 	return nil
 }
 
+// reconcileOpenShiftTrustedCAs checks for the existence of /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem, if it exists,
+// creates a new ConfigMap to be mounted in the CPO deployment utilizing the file
+func (r *HostedClusterReconciler) reconcileOpenShiftTrustedCAs(ctx context.Context, hostedControlPlane *hyperv1.HostedControlPlane) (bool, error) {
+	trustedCABundle := new(bytes.Buffer)
+	var trustCABundleFile []byte
+	if _, err := os.Stat("/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"); err == nil {
+		trustCABundleFile, err = os.ReadFile("/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem")
+		if err != nil {
+			return false, fmt.Errorf("unable to read trust bundle file: %w", err)
+		}
+	} else if err != nil && errors.Is(err, os.ErrNotExist) {
```

Comment:
For the sake of easier reviews and readability please let's try to avoid nested if/else as much as possible.

this could be simplified a bit with
```
	if err == nil {
		trustCABundleFile, err = os.ReadFile("/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem")
		if err != nil {
			return false, fmt.Errorf("unable to read trust bundle file: %w", err)
		}
	}
	if err != nil {
		if errors.Is(err, os.ErrNotExist) {
			return false, nil
		}
		
		return false, err
	}
```